frontend/src/views/dev_frame/load-mock-data.js

```diff
@@ -0,0 +1,227 @@
+import { normalizeImport } from "../../libs/import-export";
+import {
+    instructorsSelector,
+    upsertApplicant,
+    upsertAssignment,
+    upsertPosition,
+    upsertSession,
+    upsertInstructor,
+    upsertContractTemplate,
+    fetchSessions,
+    setActiveSession,
+} from "../../api/actions";
+
+import {
+    positionSchema,
+    applicantSchema,
+    assignmentSchema,
+} from "../../libs/schema";
+import React from "react";
+import { useSelector } from "react-redux";
+import { Button, Modal, ProgressBar } from "react-bootstrap";
+import mockSessionData from "../../mock_data/session.json";
+import mockContractTemplate from "../../mock_data/contract_template.json";
+import mockInstructors from "../../mock_data/instructor.json";
+import mockApplicantData from "../../mock_data/applicant.json";
+import mockPositionData from "../../mock_data/position.json";
+import mockAssignmentData from "../../mock_data/assignment.json";
+import { useThunkDispatch } from "../../libs/thunk-dispatch";
+import { prepareFull } from "../../libs/import-export";
+
+/**
+ * A button to automatically set up a mock session with contract_template and
+ * instructors, and upsert positions, applicants, and assignments from mock data
+ * JSON files in order.
+ *
+ * This component only renders when `process.env.REACT_APP_DEV_FEATURES` is truthy.
+ *
+ * @export
+ * @returns {React.ElementType}
+ */
+function LoadMockButton() {
+    const [disable, setDisable] = React.useState(false);
+    const [inProgress, setInProgress] = React.useState(false);
+    const [stage, setStage] = React.useState("");
+    const [now, setNow] = React.useState(0);
+    const dispatch = useThunkDispatch();
+    const instructors = useSelector(instructorsSelector);
+    let session;
+    let contractTemplates = [];
+    let positions = [];
+    let applicants = [];
+    let count;
+    let total;
+
+    async function setupMockSession() {
+        try {
+            setStage("Session");
+
+            // create the mock session
+            session = await dispatch(upsertSession(mockSessionData));
+            await dispatch(setActiveSession(session));
+
+            setStage("Contract Template");
+            const currContractTemplate = await dispatch(
+                upsertContractTemplate(mockContractTemplate)
+            );
+            contractTemplates.push(currContractTemplate);
+            setNow(50);
+            // upsert any missing instructors
+            setStage("Instructors");
+            for (let instructor of mockInstructors) {
+                if (!instructors.some((i) => i.utorid === instructor.utorid)) {
+                    const newInstructor = await dispatch(
+                        upsertInstructor(instructor)
+                    );
+                    instructors.push(newInstructor);
```

Comment:
This shouldn't be needed. When the instructors update the selector should automatically update.

Reply:
Yes that's what I originally thought too... but I checked and it did not update `instructors`

Reply:
I suspect that's a bug, then...